src/Providers/Contracts/ProviderInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Contracts;
+
+use InvalidArgumentException;
+use WordPress\AiClient\Providers\DTO\ProviderMetadata;
+use WordPress\AiClient\Providers\Models\Contracts\ModelInterface;
+use WordPress\AiClient\Providers\Models\DTO\ModelConfig;
+
+/**
+ * Interface for AI providers.
+ *
+ * Providers represent AI services (Google, OpenAI, Anthropic, etc.)
+ * and provide access to models, metadata, and availability information.
+ *
+ * @since n.e.x.t
+ */
+interface ProviderInterface
+{
+    /**
+     * Gets provider metadata.
+     *
+     * @since n.e.x.t
+     *
+     * @return ProviderMetadata Provider metadata.
+     */
+    public static function metadata(): ProviderMetadata;
+
+    /**
+     * Creates a model instance.
+     *
+     * @since n.e.x.t
+     *
+     * @param string                          $modelId     Model identifier.
+     * @param ModelConfig|array<string,mixed> $modelConfig Model configuration.
+     * @return ModelInterface Model instance.
+     * @throws InvalidArgumentException If model not found or configuration invalid.
+     */
+    public static function model(string $modelId, $modelConfig = []): ModelInterface;
```

Comment:
I think we should revise this to only accept `ModelConfig` (even though in the architecture it is so far outlined to accept both). It'll typically be passed from the `PromptBuilder` instance anyway, so allowing an easy workaround of an array here isn't needed, and it would put an additional (albeit small) burden on the provider implementation to handle transforming it - which seems unnecessary.

We could adjust that in the architecture diagrams as well.

```suggestion
     * @param ?ModelConfig $modelConfig Model configuration.
     * @return ModelInterface Model instance.
     * @throws InvalidArgumentException If model not found or configuration invalid.
     */
    public static function model(string $modelId, ?ModelConfig $modelConfig = null): ModelInterface;
```

Reply:
Agreed! 👍